realsense2_camera/include/base_realsense_node.h

```diff
@@ -97,7 +97,8 @@ namespace realsense2_camera
         BaseRealSenseNode(rclcpp::Node& node,
                           rs2::device dev,
                           std::shared_ptr<Parameters> parameters,
-                          bool use_intra_process = false);
+                          bool use_intra_process = false,
+                          bool is_rosbag_file = false);
```

Comment:
Since you are getting the rs2::device as well, try to drop the new parameter (all over the code) and use this API, see if it's possible:
```
if (dev.is<playback>())
```

Reply:
Thanks. 
It make sense and make the code easier.
I saw the PR you talked about, but still I wanted to do this without asking the primary node about rosbag_filename parameter value, since I don't thinks this is the best way to do this.

Reply:
Great, Lets also check if this API works too:
```
sensor.is<playback_sensor>()
```

If this works we can drop more changes.
Worth checking

Reply:
Not sure it's possible, I don't see `playback_sensor` in the API